### PR TITLE
Add pup

### PR DESCRIPTION
**Pull request by CRKatri**

### CODE REVIEW

**Comment by Diatrus** [comment]
Waiting on golang amd64

**Comment by Diatrus** [comment]
Please update to use the common golang arguments

**Comment by Diatrus** [comment]
Since this project seems inactive, can you switch this to use the latest commit? (I'm sure you know the format for git commit packages) 👍

File: pup.mk [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+endif
+
+SUBPROJECTS += pup
+PUP_VERSION := 0.4.0
+DEB_PUP_V   ?= $(PUP_VERSION)
+
+pup-setup: setup
+	-[ ! -f "$(BUILD_SOURCE)/pup-$(PUP_VERSION).tar.gz" ] && \
+		wget -q -nc -O$(BUILD_SOURCE)/pup-$(PUP_VERSION).tar.gz \
+			https://github.com/ericchiang/pup/archive/v$(PUP_VERSION).tar.gz
```
Comment (TheRealKeto) [inline pup.mk:12]: Stop using this.. we have the ``GITHUB_ARCHIVE`` function for a reason.

File: pup.mk [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	cp -a $(BUILD_WORK)/pup/pup $(BUILD_STAGE)/pup/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/bin
+	touch $(BUILD_WORK)/pup/.build_complete
+endif
+
+pup-package: pup-stage
+	# pup.mk Package Structure
+	rm -rf $(BUILD_DIST)/pup
+	
+	# pup.mk Prep pup
+	cp -a $(BUILD_STAGE)/pup $(BUILD_DIST)/
```
Comment (TheRealKeto) [inline pup.mk:37]: Remove the slash after ``$(BUILD_DIST)``; it's not needed.

**Review by TheRealKeto — CHANGES_REQUESTED** [review]

**Pull request merged**